src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,595 @@
+            N (int): subsampling iterations (default 100). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            beta (double): exponent parameter (default 0.). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            C (double): constant parameter (default 10.). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+
+        Returns:
+            delta (double): estimated scale that can be used with eg agglomerative clustering.
+        """
+        num_pts = X.shape[0]
+        delta, m = 0., int(  num_pts / np.exp((1+beta) * np.log(np.log(num_pts)/np.log(C)))  )
+        for _ in range(N):
+            subpop = np.random.choice(num_pts, size=m, replace=False)
+            if self.input_type == "point cloud":
+                d, _, _ = directed_hausdorff(X, X[subpop,:])
+            if self.input_type == "distance matrix":
+                d = np.max(np.min(X[:,subpop], axis=1), axis=0)
+            delta += d/N
+        return delta
+
+    def get_optimal_parameters_for_agglomerative_clustering(self, X, beta=0., C=10., N=100):
+        """
+        Compute optimal scale and resolutions for a point cloud or a distance matrix.
+
+        Parameters:
+            X (numpy array of shape (num_points) x (num_coordinates) if point cloud and (num_points) x (num_points) if distance matrix): input point cloud or distance matrix.
+            beta (double): exponent parameter (default 0.). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            C (double): constant parameter (default 10.). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+            N (int): subsampling iterations (default 100). See http://www.jmlr.org/papers/volume19/17-291/17-291.pdf for details.
+
+        Returns:
+            delta (double): optimal scale that can be used with agglomerative clustering.
+            resolutions (numpy array of shape (num_filters): optimal resolutions associated to each filter.
+        """
+        num_filt, delta = self.filters.shape[1], 0
+        delta = self.estimate_scale(X=X, N=N, C=C, beta=beta)
+
+        pairwise = pairwise_distances(X, metric="euclidean") if self.input_type == "point cloud" else X
+        pairs = np.argwhere(pairwise <= delta)
+        num_pairs = pairs.shape[0]
+        res = []
+        for f in range(num_filt):
+            F = self.filters[:,f]
+            resf = 0
+            for p in range(num_pairs):
+                resf = max(resf, abs(F[pairs[p,0]] - F[pairs[p,1]]))
+            res.append(resf)
+
+        return delta, np.array(res)
+
+    def fit(self, X, y=None, filters=None, colors=None):
+        """
+        Fit the MapperComplex class on a point cloud or a distance matrix: compute the Mapper complex and store it in a simplex tree called simplex_tree_.
```

Comment:
(same in the 3 classes)
The trailing underscore means that sphinx interprets this as a link to some label simplex_tree
> docstring of gudhi.sklearn.cover_complex.GraphInducedComplex.fit:1: ERROR: Unknown target name: "simplex_tree".

You could avoid it with `` `simplex_tree_` `` or `simplex_tree\_` I think.

Reply:
Good catch, thanks.

Reply:
I think you only changed one, not all three?